acp/identity.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2024 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package acp
+
+import (
+	cosmosSecp256k1 "github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	"github.com/cosmos/cosmos-sdk/types"
+	"github.com/decred/dcrd/dcrec/secp256k1/v4"
+	"github.com/sourcenetwork/immutable"
+)
+
+// NoIdentity specifies an anonymous actor.
+var NoIdentity = immutable.None[Identity]()
+
+// Identity uniquely identifies an acp actor.
+type Identity interface {
```

Comment:
conflict: My branch introduces similar functionality but instead of using `Identity` as an interface and then two different structs implementing it, there is just one `Identity` struct that has three components:

``` go
// Identity holds the bits that make up an actor's unique identity.
type Identity struct {
	// An actor's private key. Uses secp256k1 algorithm.
	PrivateJWK string

	// An actor's corresponding public key address.
	PublicKey string

	// An actor's DID. Generated from the public key address.
	DID string
}
```

Note: `did` is what it will be after my stuff merges not a sourcehub address.

IMO we don't need to over abstract the identity with an interface, just a struct like above would suffice where what would have as `PrivateKeyIdentity` would just be the `Identity` struct above with all non-empty fields, and `PublicKeyIdentity` would the `Identity` struct above with just `PrivateJWK` as empty. One liner documenting this on the struct would make it clear between these differences.

todo: unabstract interface identity implementation

Reply:
I've flattened like the identity interface to struct like you suggested. Let me know which package you would like for it to exist in.